mypy/messages.py

```diff
@@ -628,8 +628,13 @@ def incompatible_argument(self, n: int, m: int, callee: CallableType, arg_type:
             msg = 'Argument {} {}has incompatible type {}; expected {}'.format(
                 arg_label, target, self.quote_type_string(arg_type_str),
                 self.quote_type_string(expected_type_str))
-            if isinstance(arg_type, Instance) and isinstance(expected_type, Instance):
-                notes = append_invariance_notes(notes, arg_type, expected_type)
+            expected_types = (
+                expected_type.items if isinstance(expected_type, UnionType)
+                else [expected_type]
+            )
```

Comment:
Please rewrite this as an if/else *statement*. The long-form conditional expression is less clear than that.

Reply:
Thanks, it should be better now.